doc/examples/big_data_ingestion.py

```diff
@@ -0,0 +1,146 @@
+
+In particular, we will show you:
+
+* How to build a shuffle ingestion pipeline that loads, shuffles and feeds data
+  into distributed trainers in few lines of code;
+* How to scale the pipeline from ingesting 100MB data to
+  500GB data.
+
+.. image:: ../data/dataset-repeat-2.svg
+    :align: center
+
+"""
+
+###############################################################################
+# Python Setup
+# ------------
+#
+# First, we'll import all the libraries we'll be using. This step also helps us
+# verify that the environment is configured correctly. If any of the imports
+# are missing, an exception will be raised.
+
+import ray
+
+from typing import List
+from ray.data.dataset_pipeline import DatasetPipeline
+
+#######################################################################
+# Build shuffle ingestion pipeline
+# ----------------------------------
+#
+# A typical machine learning ingestion pipeline consists of the following 4
+# steps:
+#
+# 1. Load the training data from external storage;
+# 2. Iterate the data for multiple epochs;
+# 3. In each epoch, applying global shuffle to normalize the data;
+# 4. In each epoch, split the shuffled data into shards, and feed shards to
+#    distributed trainers;
+#
+# Let’s see how we implement such pipeline using Ray Dataset:
+
+
+def create_shuffle_pipeline(training_data_dir: str, num_epochs: int,
+                            num_shards: int) -> List[DatasetPipeline]:
+
+    return ray.data.read_parquet(training_data_dir)\
+        .repeat(num_epochs)\
+        .random_shuffle()\
+        .split(num_shards, equal=True)
+
+
+############################################################################
+# We’ve now defined a ``create_shuffle_pipeline`` function that creates an
+# ingestion pipeline.
+# It reads ``training_data_dir``, iterates for ``num_epochs`` times,
+# where in each epoch it
+# shuffles and splits the result into ``num_shards``.
+
+###############################################################################
+# Feed the pipeline into trainers
+# -----------------------------------
+# Let’s also implement a ``TrainingWorker`` which consumes the shuffled data
+# from each shard.
+#
+# For simplicity, we will define a
+# `Ray Actor <https://docs.ray.io/en/latest/actors.html>`_ that emulates
+# training workers. Specifically,
+#
+# 1. It takes one shard of the shuffle pipeline for training;
+# 2. It iterates the shard to get per epoch training dataset;
+# 3. It then consumes the dataset by batches;
+
+
+@ray.remote
+class TrainingWorker:
+    def __init__(self, rank: int, shard: DatasetPipeline):
+        self.rank = rank
+        self.shard = shard
+
+    def train(self):
+        for epoch, training_dataset in enumerate(self.shard):
+            # Following code emulates epoch based SGD training.
+            # TODO: replace the code for real training.
+            for i, batch in enumerate(training_dataset.iter_batches()):
```

Comment:
Might want to give a concrete batch size here since `iter_batches()` will return full blocks by default, which will most likely be skewed.
```suggestion
            for i, batch in enumerate(training_dataset.iter_batches(batch_size=25000)):
```